packages/db/tests/search/test-AsyncOperationSearch.js

```diff
@@ -0,0 +1,331 @@
+'use strict';
+
+const test = require('ava');
+const cryptoRandomString = require('crypto-random-string');
+const { v4: uuidv4 } = require('uuid');
+const omit = require('lodash/omit');
+const range = require('lodash/range');
+const { AsyncOperationSearch } = require('../../dist/search/AsyncOperationSearch');
+const {
+  translatePostgresAsyncOperationToApiAsyncOperation,
+} = require('../../dist/translate/async_operations');
+
+const {
+  destroyLocalTestDb,
+  generateLocalTestDb,
+  fakeAsyncOperationRecordFactory,
+  migrationDir,
+  AsyncOperationPgModel,
+} = require('../../dist');
+
+const testDbName = `asyncOperation_${cryptoRandomString({ length: 10 })}`;
+
+test.before(async (t) => {
+  const { knexAdmin, knex } = await generateLocalTestDb(
+    testDbName,
+    migrationDir
+  );
+
+  t.context.knexAdmin = knexAdmin;
+  t.context.knex = knex;
+
+  t.context.asyncOperationPgModel = new AsyncOperationPgModel();
+  t.context.asyncOperations = [];
+  t.context.asyncOperationSearchTmestamp = 1579352700000;
+
+  range(100).map((num) => (
+    t.context.asyncOperations.push(fakeAsyncOperationRecordFactory({
+      cumulus_id: num,
+      updated_at: new Date(t.context.asyncOperationSearchTmestamp + (num % 2)),
+      operation_type: num % 2 === 0 ? 'Bulk Granules' : 'Data Migration',
+      task_arn: num % 2 === 0 ? cryptoRandomString({ length: 3 }) : undefined,
+    }))
+  ));
+
+  await t.context.asyncOperationPgModel.insert(
+    t.context.knex,
+    t.context.asyncOperations
+  );
+});
+
+test.after.always(async (t) => {
+  await destroyLocalTestDb({
+    ...t.context,
+    testDbName,
+  });
+});
+
```

Comment:
Generic: these tests all look really solid, but do we care about certain permutations of test fields? 

e.g. "AsyncOperationsSearch supports infix and prefix in the same query"?      Are we sure that using all or most of them in combination does what we expect?

Reply:
We have a test for rather complex search here:
https://github.com/nasa/cumulus/blob/feature/es-phase-2/packages/db/tests/search/test-GranuleSearch.js#L392

Reply:
Ok, but that's testing the granules subclass, which can have different implementations than the base class...
It's probably reasonable to have generic tests against the base class, but it feels worrysome to not have units covering the combined methods for modified subclasses if we think it's possible for the individual query methods to interfere with each other.  

I'm not convinced that's entirely the case here, given infix/prefix is a where clause modifier, but we should consider this pattern for other objects and/or the base class.